documentation/proton.html

```diff
@@ -123,41 +123,76 @@ <h3 id="sub-databases">Sub-databases</h3>
 
 <h2 id="transaction-log">Transaction log</h2>
 <p>
-Content nodes have a transaction log to persist mutating operations immediately.
-ToDo: When synced to file system?
+Content nodes have a transaction log to persist mutating operations.
+The transaction log persists operations by file append.
+Having a transaction log simplifies proton's in-memory index structures
+and enables steady-state high performance, read more below.
 </p><p>
 Operations are written immediately,
 irrespective of <a href="reference/services-content.html#visibility-delay">visibility-delay</a>.
+Operations are not <em>sync'ed</em> to the file system -
+hence, writes are <em>not guaranteed</em> to exist in the transaction log.
+Flush jobs (below) sync the transaction log, for consistency.
 </p><p>
 Default, proton will
 <a href="reference/services-content.html#flush-on-shutdown">flush components</a>
 like attribute vectors and memory index on shutdown, for quicker startup.
-</p><p>
 </p>
 
 
 
 <h2 id="index">Index</h2>
 <p>
-For all <a href="reference/search-definitions-reference.html#index">indexed fields</a>,
-proton has a memory index for the recent changes, which is flushed to a disk-based index -
-see <a href="#memory-index-flush">memory-index-flush</a>.
-Disk-based indexes are subsequently merged by <a href="#disk-index-fusion">disk index fusion</a>.
-Index fields are string fields and are used for text search.
+Index fields are string fields, used for text search.
 Other field types are <a href="attributes.html">attributes</a>.
+</p>
+
+
+<h3 id="updates">Updates</h3>
+<p>
+For all <a href="reference/search-definitions-reference.html#index">indexed fields</a>,
+proton has a memory index for the recent changes, implemented using a B-tree.
+This is periodically <a href="#memory-index-flush">flushed</a> to a disk-based posting list index.
+Disk-based indexes are subsequently <a href="#disk-index-fusion">merged</a>.
+</p><p>
+This means that recent changes are in-memory B-tree changes,
```

Comment:
I don't understand the semantics of this sentence. Consider rewrite: "Updating the in-memory B-trees is lock-free, implemented using copy-on-write semantics". That the changes are recent is already mentioned in the previous paragraph.

Reply:
agree